gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -228,6 +236,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                     )
                 }
             )
+            kspTask.commandLineArgumentProviders.addAll(kspExtension.commandLineArgumentProviders)
```

Comment:
Because it's unsafe to call `asArgument()` at configuration time, I guess we need to wrap this with project.provider like `kspTask.options` in the lines above. (Or, is `CommandLineArgumentProvider` already considered lazy values? If so, LGTM)
```
kspTask.commandLineArgumentProviders.addAll(
    kspTask.project.provider {
        kspExtension.commandLineArgumentProviders
    }
)
```